### DISCUSSION TITLE
Can I reset a submap after a subkey?

### DISCUSSION THREAD

**Discussion by txtyash** [hyprwm/Hyprland#2546] [Q&A] [answered]
So this is the way to reset a submap in the docs(by binding a key to reset it):
```
# sets repeatable binds for resizing the active window
binde=,right,resizeactive,10 0
binde=,left,resizeactive,-10 0
binde=,up,resizeactive,0 -10
binde=,down,resizeactive,0 10

# use reset to go back to the global submap
bind=,escape,submap,reset 
```
I want to exit the submap as soon as I press a subkey and it gets executed. I'm creating a submap for taking screenshots. So I want to press `mod+s` for starting it then press `f` for fullscreen screenshot and exit the submap without having to press `escape`

**Comment by ghost** [comment]
Is that what you’re asking for ? Just set `bind = , F, submap, reset` after the f command
> You can also set the same keybind to perform multiple actions, such as resize and close the submap, like so:
```
bind=ALT,R,submap,resize

submap=resize

bind=,right,resizeactive,10 0
bind=,right,submap,reset
# ...

submap=reset
```
> This works because the binds are executed in the order they appear, and assigning multiple actions per bind is possible.

**Reply by txtyash** [reply]
I didn't work for me:
```conf
# start screenshot submap 
bind = $mainMod, s, submap, screenshot 

submap = screenshot

# repeatable binds for screenshot submap
binde= , space, exec, grim - -g "0,0 1920x1080" $(xdg-user-dir)/screenshots/full/$(date +'%F-%T.png') | wl-copy, reset
binde= , s    , exec, grim - -g "$(slurp)" $(xdg-user-dir)/screenshots/select/$(date +'%F-%T.png') | wl-copy, reset

submap = reset
```
Neither with `binde` or `bind`

**Reply by vaxerski** [reply, marked as answer]
thats a command so append `; hyprctl dispatch submap reset` instead of `, reset` which is invalid bash syntax
